src/NodeApi/JSReference.cs

```diff
@@ -118,9 +118,19 @@ protected virtual void Dispose(bool disposing)
         {
             IsDisposed = true;
             napi_ref handle = _handle; // To capture in lambda
-            _context.SynchronizationContext.Post(
-                () => napi_delete_reference((napi_env)_context, handle).ThrowIfFailed(),
-                allowSync: true);
+
+            // The context may be null if the reference was creatd from a "no-context" scope such
+            // as the native host. In that case the reference must be disposed from the JS thread.
+            if (_context == null)
+            {
+                napi_delete_reference((napi_env)JSValueScope.Current, handle).ThrowIfFailed();
```

Comment:
We should verify that we are in the JS thread at this point. Could you add a TODO?

Reply:
There is already a check: `JSValueScope.Current` will throw a "no current scope" exception.